### PR TITLE
Allergies on FQS

### PR DESCRIPTION
**Pull request by pmadria3**
This PR is wiring Allergies onto FQS. 
Screenshot 2023-06-06 at 3 51 43 PM

### CODE REVIEW

File: src/fhir/allergies.ts [hunk 1]
```diff
@@ -1,48 +1,67 @@
...
+              upid: patient.UPID,
+              cursor: "",
+              first: 1000,
+              sort: {
+                lastUpdated: "DESC",
+              },
+            })) as AllergyGraphqlResponse;
+
+            const nodes = data.AllergyIntoleranceConnection.edges.map((x) => x.node);
+            // TODO: No longer needed once https://zeushealth.atlassian.net/browse/DRT-249 is resolved.
```
Comment (sarahmangels) [inline src/fhir/allergies.ts:31]: which line is this TODO for?

File: src/fhir/allergies.ts [hunk 2]
```diff
@@ -1,48 +1,67 @@
...
+
+            const results = applyAllergyFiltersFQS(nodes);
+            if (results.length === 0) {
+              Telemetry.countMetric("req.count.allergies.none");
```
Comment (sarahmangels) [inline src/fhir/allergies.ts:35]: do you want to add the `["fqs"]` tag thingy like this
https://github.com/zeus-health/ctw-component-library/blob/5717f890f9583e67c024b93794afe0bdb4835aaa/src/services/conditions.ts#L44

File: src/fhir/allergies.ts [hunk 3]
```diff
@@ -1,48 +1,67 @@
...
+            }
+            Telemetry.histogramMetric("req.count.allergies", results.length);
```
Comment (sarahmangels) [inline src/fhir/allergies.ts:37]: same here

File: src/fhir/allergies.ts [hunk 4]
```diff
@@ -1,48 +1,67 @@
...
+            return results;
+          } catch (e) {
+            throw new Error(`Failed fetching allergies information for patient ${patient.UPID}`);
           }
-        );
+        }, "req.timing.allergies")
```
Comment (sarahmangels) [inline src/fhir/allergies.ts:42]: and/or here?

File: src/fhir/models/allergies.ts [hunk 5]
```diff
@@ -59,6 +59,11 @@ export class AllergyModel extends FHIRModel<fhir4.AllergyIntolerance> {
     return organizationDisplay?.managingOrganization?.display || organizationName;
   }
 
+  get managingOrganizationFQS(): string | undefined {
```
Comment (sarahmangels) [inline src/fhir/models/allergies.ts:62]: is this used anywhere?

Comment (pmadria3) [reply src/fhir/models/allergies.ts:62]: Yep the table (specified in the columns for the table) we need to specify since the object is mutated from FHIR and breaks the drawer since managingOrg isn't retrieved the same in FQS vs ODS

**Review by sarahmangels — APPROVED** [review]

File: src/fhir/models/allergies.ts (hunk 5 above)
Comment (WesleyKapow) [inline src/fhir/models/allergies.ts:62]: Instead of introducing a new method which pushes the complexity to consumers, we can encapsulate this complexity. Modify `managingOrganization` to "do the right thing". It can see if we have the resource provided from FQS. In fact, you might be able to modify the `findReference` helper to look on the passed `reference` object for these FQS `resource` keys.

File: src/components/content/allergies/story-helpers/mocks/allergy-intoleranceFQS.ts [hunk 6]
```diff
@@ -0,0 +1,638 @@
+export const allergyIntoleranceFQS = {
```
Comment (WesleyKapow) [inline src/components/content/allergies/story-helpers/mocks/allergy-intoleranceFQS.ts:1]: nit: We don't use capitals in our file names.. this should be `allergy-intolerance-fqs.ts`

File: src/components/content/allergies/helpers/allergies-filter.tsx [hunk 7]
```diff
@@ -18,4 +18,16 @@ export const applyAllergyFilters = (
   return allergyData;
 };
 
+export const applyAllergyFiltersFQS = (data: fhir4.AllergyIntolerance[]) => {
```
Comment (WesleyKapow) [inline src/components/content/allergies/helpers/allergies-filter.tsx:21]: Why is this needed? Seems almost identical to the above. If it's due to not having a `includedResources` map, you can mark that as optional on the above and then use it.

File: .changeset/heavy-ways-fold.md [hunk 8]
```diff
@@ -0,0 +1,5 @@
+---
+"@zus-health/ctw-component-library": minor
+---
+
+Update Allergies to utilize FQS.
```
Comment (WesleyKapow) [inline .changeset/heavy-ways-fold.md:5]: Should indicate this is guarded by a prop. As in, this note as written reads like allergies now uses FQS which is not true.

> Allergies can now read from FQS with the new `enableFQS` prop.
